sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeedSource.cs

```diff
@@ -38,7 +38,7 @@ dataFeedDetail switch
                 MySqlDataFeed d => new MySqlDataFeedSource(d.DataSourceParameter),
                 PostgreSqlDataFeed d => new PostgreSqlDataFeedSource(d.DataSourceParameter),
                 SQLServerDataFeed d => new SqlServerDataFeedSource(d.DataSourceParameter, d.AuthenticationType, d.CredentialId),
-                _ => throw new InvalidOperationException("Invalid DataFeedDetail type")
+                _ => new UnknownDataFeedSource(dataFeedDetail.DataSourceType)
```

Comment:
We have three models for this scenario:
- A public model, exposed to users. This is the current class, `DataFeedSource`. Let's call it "public model".
- An internal model, used by the service. This internal model (`DataFeedDetail`) is used for Get, List, and Create operations. Let's call it "internal model".
- Another internal model, used by the service. This internal model (`DataFeedDetailPatch`) is used for Update operations. Let's call it "internal update model".

In conversions between these models, we were just throwing when an unknown type was found. Now we're handling this properly.

This is the "internal model" => "public model" conversion.